lib/core/AxiosHeaders.js

```diff
@@ -115,6 +115,10 @@ Object.assign(AxiosHeaders.prototype, {
       utils.forEach(header, (_value, _header) => {
         setHeader(_value, _header, valueOrRewrite);
       });
+    } else if (header instanceof AxiosHeaders) {
```

Comment:
You can also add the condition in the above if

```javascript
if (utils.isPlainObject(header) || header instanceof AxiosHeaders) {
  utils.forEach(header, (_value, _header) => {
    setHeader(_value, _header, valueOrRewrite);
  });
} else {
  setHeader(valueOrRewrite, header, rewrite);
}
```
In order to not maintain 2 for loops

Great job anyway !

Reply:
Thanks @paulsouche, I've updated this now